error.go

```diff
@@ -107,6 +107,16 @@ func (nnr NetworkNameError) Error() string {
 // Forbidden denotes the type of this error
 func (nnr NetworkNameError) Forbidden() {}
 
+// PredefinedNetworkExistError is returned when user tries to create predefined network that already exists.
+type PredefinedNetworkExistError string
```

Comment:
If it is a predefined network, why would will be ever created by the user?

Reply:
Moby will try to create all networks blindly at: https://github.com/moby/moby/pull/36316/files#diff-755660d548f37a258383def7d94b039aR153

So for something like `docker service create --network=host nginx`, it will attempt to create the `host` network first and then skip upon detecting a failure.

Reply:
isn't it better to check if a network is a predefined instead of leveraging an error for this job?

Reply:
Before we create network , we do check and respond to the caller saying network already exists.  only difference is we do allow or continue if user defined network is getting exists already but we try to return error in case of predefined network. Just had lengthy discussion with @ddebroy :)